### PR TITLE
Longevity testing all stages

### PR DESCRIPTION
**Pull request by PrasadDesala**
In Longevity testing we make sure that the ODF can continue
responding to the user and admin operations reliably and consistently under
significant load for an extended period of time.

To achieve this the tests are categorized into below stages.

Stage-0: Creates the initial softconfiguration of all supported types
on the Longevity cluster and keep the created resources stay forever on the cluster
Stage-1: Configure and run OCP and Application workloads. These workloads
will be run continuously for a specified period of time
Stage-2: Sequential PVC, OBC, APP pod lifecycle operations for a specified period of time
Stage-3: Concurrent PVC, OBC, APP pod lifecycle operations for a specified period of time
Stage-4: Clone, Snapshot, Expand operations for a specified period of time

This test starts all stages of Longevity testing

Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

**Comment by ebenahar** [comment]
@PrasadDesala is the intention to run all stages concurrently with this change?

**Comment by PrasadDesala** [comment]
> @PrasadDesala is the intention to run all stages concurrently with this change?
@ebenahar The function is made compatible to run both sequential and concurrent execution of Longevity stages. This is developed this way keeping system tests and OMADops in mind. For Longevity as per the current test plan we would be running the stages in sequential manner (stage1 will be running throughout all other stages) and for system tests and OMADops we can start all the stages concurrently in background.  Since we are running concurrent operations of  app workloads, resource creation/deletions, sequential and concurrent resource operations, snapshot, clone, pvc expansion, it can create a good amount of load on the cluster and also we will be matching the real world scenario. Achieving stability of these tests is of course a bit difficult and challenging in the initial runs as it involves many concurrent background operations but should be possible after repeated dry runs.

**Comment by ebenahar** [comment]
@PrasadDesala thank you Prasad for the expolanation. It's clear now

**Review by ebenahar — DISMISSED** [review]

**Review by Akarsha-rai — DISMISSED** [review]

File: ocs_ci/ocs/longevity.py [hunk 1]
```diff
@@ -1137,6 +1142,115 @@ def stage_4(
...
+                ),
+                ThreadPoolExecutor(max_workers=1).submit(
+                    self.stage_2,
+                    project_factory,
+                    multi_pvc_pod_lifecycle_factory,
+                    multi_obc_lifecycle_factory,
+                    run_time=stage_run_time,
+                ),
+                ThreadPoolExecutor(max_workers=1).submit(
+                    self.stage3,
```
Comment (vavuthu) [inline ocs_ci/ocs/longevity.py:1198]:
```suggestion
                    self.stage_3,
```

Comment (PrasadDesala) [reply ocs_ci/ocs/longevity.py]: Done

**Review by vavuthu — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Pull request merged**